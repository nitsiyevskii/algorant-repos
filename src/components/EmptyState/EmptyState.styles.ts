import { createStyles } from "src/utils/createStyles";

export const useStyles = createStyles((theme) => ({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    paddingHorizontal: 32,
  },
  message: {
    fontSize: theme.fonts.size.lg,
    color: theme.colors.text.secondary,
    textAlign: "center",
    marginTop: 16,
    fontWeight: theme.fonts.weight.medium,
  },
  description: {
    fontSize: theme.fonts.size.md,
    color: theme.colors.text.tertiary,
    textAlign: "center",
    marginTop: 8,
  },
}));

