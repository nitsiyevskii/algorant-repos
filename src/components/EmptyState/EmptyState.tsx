import React from 'react';
import { View, Text } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { useTheme } from 'src/hooks/useTheme';
import { useStyles } from './EmptyState.styles';

interface EmptyStateProps {
  icon?: keyof typeof Ionicons.glyphMap;
  message: string;
  description?: string;
}

export const EmptyState: React.FC<EmptyStateProps> = ({
  icon = 'search',
  message,
  description,
}) => {
  const theme = useTheme();
  const styles = useStyles();

  return (
    <View style={styles.container}>
      <Ionicons name={icon} size={64} color={theme.colors.text.tertiary} />
      <Text style={styles.message}>{message}</Text>
      {description && <Text style={styles.description}>{description}</Text>}
    </View>
  );
};

