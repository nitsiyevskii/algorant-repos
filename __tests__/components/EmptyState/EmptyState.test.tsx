import { render } from '@testing-library/react-native';
import { EmptyState } from 'src/components/EmptyState/EmptyState';

describe('EmptyState', () => {
  it('renders correctly with message', () => {
    const { getByText } = render(
      <EmptyState message="No data available" />
    );
    
    expect(getByText('No data available')).toBeTruthy();
  });

  it('renders with custom icon', () => {
    const { getByText } = render(
      <EmptyState icon="folder-open" message="Empty folder" />
    );
    
    expect(getByText('Empty folder')).toBeTruthy();
  });

  it('renders with description', () => {
    const { getByText } = render(
      <EmptyState
        message="No results"
        description="Try adjusting your search"
      />
    );
    
    expect(getByText('No results')).toBeTruthy();
    expect(getByText('Try adjusting your search')).toBeTruthy();
  });

  it('renders without description when not provided', () => {
    const { getByText, queryByText } = render(
      <EmptyState message="No data" />
    );
    
    expect(getByText('No data')).toBeTruthy();
    expect(queryByText('Try adjusting your search')).toBeNull();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <EmptyState
        icon="search"
        message="No repositories found"
        description="Try adjusting your search"
      />
    );
    
    expect(toJSON()).toMatchSnapshot();
  });
});

