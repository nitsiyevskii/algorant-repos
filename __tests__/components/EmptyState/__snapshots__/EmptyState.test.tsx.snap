// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 32,
    }
  }
>
  <Text
    color="#8E8E93"
    name="search"
    size={64}
    testID="icon"
  >
    search
  </Text>
  <Text
    style={
      {
        "color": "#3C3C43",
        "fontSize": 18,
        "fontWeight": "500",
        "marginTop": 16,
        "textAlign": "center",
      }
    }
  >
    No repositories found
  </Text>
  <Text
    style={
      {
        "color": "#8E8E93",
        "fontSize": 16,
        "marginTop": 8,
        "textAlign": "center",
      }
    }
  >
    Try adjusting your search
  </Text>
</View>
`;
